frontend/packages/ceph-storage-plugin/integration-tests-cypress/tests/multiple-storageclass-selection-scenarios.ts

```diff
@@ -0,0 +1,117 @@
+  cy.byTestID(`${scName}-link`).click();
+  cy.byTestID('confirm-action').click();
+};
+
+describe('Add capacity using multiple storage classes', () => {
+  const beforeCapacityAddition = {
+    deviceSets: null,
+    portability: null,
+    devicesCount: null,
+  };
+  before(() => {
+    cy.login();
+    cy.visit('/');
+    cy.install();
+    cy.exec(`echo '${JSON.stringify(testEbsSC)}' | kubectl apply -f -`);
+    cy.exec(`echo '${JSON.stringify(testNoProvisionerSC)}' | kubectl apply -f -`);
+    commonFlows.navigateToOCS();
+    cy.byLegacyTestID('horizontal-link-Storage Cluster').click();
+  });
+  after(() => {
+    cy.exec(`echo '${JSON.stringify(testEbsSC)}' | kubectl delete -f -`);
+    cy.exec(`echo '${JSON.stringify(testNoProvisionerSC)}' | kubectl delete -f -`);
+    cy.logout();
+  });
+
+  describe('Add capacity with a new storage class having EBS as provisioner', () => {
+    const { name: scName } = testEbsSC.metadata;
+    let index: number;
+    let deviceSets: DeviceSet[];
+    before(() => {
+      // var json: K8sResourceKind
+      fetchStorageClusterJson().then((res) => {
+        const json: K8sResourceKind = JSON.parse(res.stdout);
+        beforeCapacityAddition.deviceSets = json.spec.storageDeviceSets.length;
+        addCapacity(json.metadata.uid, scName);
+        fetchStorageClusterJson().then((newRes) => {
+          const latestJson: K8sResourceKind = JSON.parse(newRes.stdout);
+          deviceSets = latestJson.spec.storageDeviceSets;
+          index = getCurrentDeviceSetIndex(deviceSets, scName);
+        });
+      });
+    });
+    it('New device set is created', () =>
+      expect(deviceSets.length).toBe(beforeCapacityAddition.deviceSets + 1));
+    it('Device count is 1 in the new device set', () => expect(deviceSets[index].count).toBe(1));
+    it('Osd portability is enabled in the new device set', () =>
+      expect(deviceSets[index].portable).toBe(true));
+  });
+
+  describe('Add capacity with an existing storage class having EBS as provisioner', () => {
+    const { name: scName } = testEbsSC.metadata;
+    let latestDeviceSets: DeviceSet[];
+    let latestIndex: number;
+    before(() => {
+      fetchStorageClusterJson().then((res) => {
+        const json: K8sResourceKind = JSON.parse(res.stdout);
+        const deviceSets: DeviceSet[] = json.spec.storageDeviceSets;
+        const index = getCurrentDeviceSetIndex(deviceSets, scName);
+        beforeCapacityAddition.deviceSets = deviceSets.length;
+        beforeCapacityAddition.portability = deviceSets[index].portable;
+        beforeCapacityAddition.devicesCount = deviceSets[index].count;
+        addCapacity(json.metadata.uid, scName);
+        fetchStorageClusterJson().then((newRes) => {
+          const latestJson: K8sResourceKind = JSON.parse(newRes.stdout);
+          latestDeviceSets = latestJson.spec.storageDeviceSets;
+          latestIndex = getCurrentDeviceSetIndex(latestDeviceSets, scName);
+        });
```

Comment:
This block is same in all cases, better to move it to a function.

Reply:
Done... Please take a look...